src/Framework/AspNetCoreAnalyzers/test/Dependencies/AddPackageTests.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.AddPackage;
+using Moq;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Verifiers.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.WebApplicationBuilder.WebApplicationBuilderAnalyzer,
+    Microsoft.AspNetCore.Analyzers.Dependencies.AddPackagesTest.MockAddPackageFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers.Dependencies;
+
+/// <remarks>
+/// These tests don't assert the fix is applied, since it takes a dependency on the internal
+/// VS-specific `PackageInstallerService`. However, the fixer is invoked in these codepaths
+/// so we can validate that the symbol resolution and checks function correctly.
+/// </remarks>
+public class AddPackagesTest
+{
+    [Fact]
+    public async Task CanFixMissingExtensionMethodForDI()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.{|CS1061:AddOpenApi|}();
+";
+
+        // Assert
+        MockAddPackageFixer.Invoked = false;
```

Comment:
This only reliably works because it's used in the same test class, so tests don't run in parallel right? Should probably mention that as a comment on the class definition so we don't accidentally use it across classes without some thought.

Reply:
Added a remark here.